### DISCUSSION TITLE
Why does zpool import <my-pool> cause /var/log, /var/lib, /var/spool to be unmounted?

### DISCUSSION THREAD

**Discussion by MadProbe** [openzfs/zfs#16154] [Q&A] [answered]
The full story:
I wanted to install Linux Mint with root as zfs pool, booted into the installer, built zfs 2.2.3 from source and used these binaries from that point onward, and followed the installer guide for Ubuntu 22.0 root on zfs, and I originally have run into the issue that at some point browser suddenly could not find a server, but I still could see my wifi working.
After some time and reboots I have finally learned that importing zpool caused flaky browser behavior: sites I have already visited are working but every other firefox didn't manage to find server, and some time later even visited sites have succumbed to this issue.
Exporting imported pools fixed the issue, but after some digging in the /var/log I have found that on pool import some directories were unmounted, though initially I hadn't realized that that meant anything.
Aswell I have found that I could still ping dns servers and any other valid ip address & localhost when I as a pool imported.
Then I have imported a pool and wanted to check /var/log/syslog again, but nano showed nothing and that this file i yet to be created: then I realized that zpool for some reason unmounted /var/log, and with it /var/lib with NetworkManager: that's why I had browser broken.
How can I continue with linux install onto zfs pool as root if I cannot use debootstrap without internet?

**Comment by IvanVolosyuk** [comment, marked as answer]
Not sure I understand the full detail of the problem, but when I setup a new system I usually do that in a chroot environment. You can import the pool with altroot using: ```zpool import -R /mnt/newsystem pool```. Next, you can bind mount whatever is needed for the root of the new system:
```
mount -t tmpfs -o size=2G none $ROOT/var/tmp
mount -t tmpfs -o size=2G none $ROOT/tmp
mount -o bind /dev $ROOT/dev
mount -o bind /sys $ROOT/sys
mount -o bind /proc $ROOT/proc
mount -o rw,nosuid,noexec,relatime,gid=5,mode=620,ptmxmode=000 devpts $ROOT/dev/pts -t devpts
mount -t tmpfs -o size=2G,mode=01777 none $ROOT/dev/shm
```
You can copy ```/etc/resolv.conf``` to ```$ROOT/etc/resolv.conf``` for DNS resolver to work.Then simply: ```cd $ROOT; chroot $ROOT bin/bash```
Sorry, if I completely missed the point.

**Reply by MadProbe** [reply]
Thank you very much for help and I have realized that when I import r/bpool I also need to specify `-R /mnt/...` again (the issue ran away in its entirety).
I for some reason thought that it would remember mountpoint from -R parameter when I import the pool again
